controllers/component_controller.go

```diff
@@ -189,8 +189,13 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 		}
 	} else {
 		if hasApplication.Status.Devfile != "" && (forceGenerateGitopsResource || len(component.Status.Conditions) > 0 && component.Status.Conditions[len(component.Status.Conditions)-1].Status == metav1.ConditionTrue && containsString(component.GetFinalizers(), compFinalizerName)) {
-			// only attempt to finalize and update the gitops repo if an Application is present & the previous Component status is good
+			// only attempt to finalize and update the gitops repo if an Application is present & not under deletion & the previous Component status is good
 			// A finalizer is present for the Component CR, so make sure we do the necessary cleanup steps
+			if !hasApplication.ObjectMeta.DeletionTimestamp.IsZero() {
```

Comment:
LGTM but honestly I am wondering what happens when Application Deletion Timestamp is 0 and we proceed but by the time the Component Deletion is happening; Application Deleteion Timestamp is now not 0 and is faster than a Component Delete?

Reply:
I think to be completely error proof, we may also need to block out Application Delete if Component Deletion is under process... 🤔

WDYT?

Reply:
It makes sense. A question I have is what if both `application` and `component` have `deletiontimestamp` not 0 and we ignore the deletion for both `application_controller` and `component_controller`. Will we keep retrying to delete `component` and/or `application`?

I was thinking if it makes sense to proceed only with the fix for the component deletion failure alert and if the issue still presists then to apply the same fix for the application too. WDYT?

Reply:
If both have deletiontimestamps not 0, then we only proceed with the Application...

If you think its going to be take more time to fix it, we can check this in.

/lgtm